library/general/src/lib/yast2/popup.rb

```diff
@@ -0,0 +1,318 @@
+require "yast"
+
+require "erb"
+
+Yast.import "Label"
+Yast.import "UI"
+
+module Yast2
+  # Class responsible for showing popups. It have small, but consistent API.
+  # Intended as replacement for Yast::Popup module.
+  # @note as UI is not easy to test, it is recommended after modifications to this class run
+  #   examples/popup_series_tester.sh which tests common combination of options.
+  # @note for rspec tests use popup_rspec where is helper for easier mocking that still does
+  # argument verifications.
+  class Popup
+    class << self
+      include Yast::I18n
+
+      # Number of lines for richtext: :auto to switch to richtext widget
+      LINES_THRESHOLD = 20
+
+      RICHTEXT_WIDTH = 60
+      RICHTEXT_HEIGHT = 10
+
+      # Shows popup and return which symbol of pressed button
+      # @param message [String] message to show
+      # @param details [String] hidden details that can be shown, if empty then it is nil shown
```

Comment:
Parse error: `if empty then it is nil show`

Reply:
ah, it should be `if empty then nothing is shown`. Probably I should be more explicit about behavior here. Like something like:
```
# @param details [String] if it is not empty, it adds button to show details, which after clicking open addition popup with given details. If string is empty, nothing is shown
```